entity-service/src/main/kotlin/com/egm/stellio/entity/web/EntityHandler.kt

```diff
@@ -336,24 +335,24 @@ class EntityHandler(
         val body = requestBody.awaitFirst()
         val contexts = checkAndGetContext(httpHeaders, body)
 
-        val expandedBody = expandJsonLdFragment(body, contexts) as Map<String, List<Any>>
         val expandedAttrId = expandJsonLdKey(attrId, contexts)!!
         checkAttributeIsAuthorized(expandedAttrId, entityUri, userId)
-        val expandedPayload = mapOf(expandedAttrId to expandedBody)
-        entityAttributeService.partialUpdateEntityAttribute(entityUri, expandedPayload, contexts)
 
-        val updatedEntity = entityService.getFullEntityById(entityUri, true)
-        entityEventService.publishEntityEvent(
-            AttributeUpdateEvent(
-                entityId = entityUri,
-                attributeName = attrId,
-                datasetId = getPropertyValueFromMapAsString(expandedBody, NGSILD_DATASET_ID_PROPERTY)?.toUri(),
-                operationPayload = removeContextFromInput(body),
-                updatedEntity = compactAndSerialize(updatedEntity!!, contexts, MediaType.APPLICATION_JSON),
-                contexts = contexts
-            ),
-            updatedEntity.type
-        )
+        val expandedPayload = parseAndExpandAttributeFragment(attrId, body, contexts)
+
+        val updateResult = entityAttributeService.partialUpdateEntityAttribute(entityUri, expandedPayload, contexts)
+
+        if (updateResult.updated.isEmpty())
+            throw ResourceNotFoundException("Unknown attribute in entity $entityId")
+        else
+            entityEventService.publishPartialUpdateEntityAttributesEvents(
+                entityUri,
+                expandedPayload,
+                updateResult,
```

Comment:
you could call it directly with the `updateResult.updated` list

Reply:
Done